mods/settings.md

```diff
@@ -579,6 +579,55 @@ $execute {
 }
 ```
 
+Incase you want to store a more complex value (e.g. a string or a custom class/struct), you'll have to implement all the required operators. This is done as follows:
+
+```cpp
+struct MyComplexSettingValue {
+    // Strings can't be used directly as they conflict with the string setting specialization
+    std::string value;
+
+    // Make sure it's comparable so that settings can check if the value has changed
+    bool operator==(MyComplexSettingValue& other) const {
+        return value == other.value;
+    }
+
+    // Optionally you can simplify the access to the value
+    operator std::string() const {
+        return value;
+    }
+
+    // Create a default implementation
+    MyComplexSettingValue() = default;
+
+    // Create a simple way to construct the value with the settings node
+    MyComplexSettingValue(std::string_view value) : value(value) {}
+
+    // Create a copy constructor
+    MyComplexSettingValue(MyComplexSettingValue& other) : value(other.value) {}
+};
+
+// You'll have to manually implement the serialization for the value
+template<>
+struct matjson::Serialize<MyComplexSettingValue> {
+    // How to serialize the value to JSON. In this case we can store it as a string

```

Comment:
`Serialize the value into JSON. In this case, we just return the value, as strings are inherently JSON-serializable`